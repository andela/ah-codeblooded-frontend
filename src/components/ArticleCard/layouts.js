export default {
  VERTICAL_LAYOUT: "VERTICAL_LAYOUT",
  HORIZONTAL_REVERSE_LAYOUT: "HORIZONTAL_REVERSE_LAYOUT",
  HORIZONTAL_LAYOUT: "HORIZONTAL_LAYOUT",
  MINIMAL_LAYOUT: "MINIMAL_LAYOUT",
  MINIMAL_AUTHOR_LAYOUT: "MINIMAL_AUTHOR_LAYOUT",
};
