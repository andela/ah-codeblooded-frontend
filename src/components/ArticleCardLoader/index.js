import ContentLoader from "react-content-loader";
import React from 'react';
import layouts, { sizes } from './layouts';

const ArticleCardLoader = ({ layout, ...props }) => (
  <ContentLoader
    rtl
    height={sizes[layout]}
    width={700}
    speed={3}
    primaryColor="#cccccc"
    secondaryColor="#ecebeb"
    {...props}
  >
    {layouts[layout]}
  </ContentLoader>
);

export default ArticleCardLoader;
