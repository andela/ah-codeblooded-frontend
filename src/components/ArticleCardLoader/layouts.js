import React from 'react';
import layouts from '../ArticleCard/layouts';

export default {
  [layouts.VERTICAL_LAYOUT]: (
    <>
      <rect x="19.03" y="76.61" rx="0" ry="0" width="458" height="120" />
      <circle cx="48.44" cy="42.02" r="24.41" />
      <rect x="83.03" y="23.61" rx="0" ry="0" width="123" height="12" />
      <rect x="84.03" y="42.61" rx="0" ry="0" width="83" height="8" />
      <rect x="21.03" y="208.61" rx="0" ry="0" width="397" height="31.05" />
      <rect x="23.03" y="255.61" rx="0" ry="0" width="287" height="6" />
      <rect x="23.03" y="275.61" rx="0" ry="0" width="281" height="6" />
    </>
  ),
  [layouts.HORIZONTAL_LAYOUT]: (
    <>
      <rect x="10.03" y="24.61" rx="0" ry="0" width="187" height="167" />
      <circle cx="246.73" cy="54.31" r="29.7" />
      <rect x="284.03" y="33.61" rx="0" ry="0" width="115" height="9" />
      <rect x="286.03" y="53.61" rx="0" ry="0" width="98" height="8" />
      <rect x="210.03" y="96.61" rx="0" ry="0" width="276" height="33" />
      <rect x="210.03" y="136.61" rx="0" ry="0" width="131" height="11" />
      <rect x="208.03" y="156.61" rx="0" ry="0" width="118" height="8" />
    </>
  ),
  [layouts.HORIZONTAL_REVERSE_LAYOUT]: (
    <>
      <rect x="304.03" y="21.61" rx="0" ry="0" width="187" height="167" />
      <circle cx="45.73" cy="56.31" r="29.7" />
      <rect x="88.03" y="37.61" rx="0" ry="0" width="115" height="9" />
      <rect x="89.03" y="55.61" rx="0" ry="0" width="98" height="8" />
      <rect x="22.03" y="97.61" rx="0" ry="0" width="276" height="33" />
      <rect x="23.03" y="138.61" rx="0" ry="0" width="131" height="11" />
      <rect x="24.03" y="156.61" rx="0" ry="0" width="118" height="8" />
    </>
  ),
  [layouts.MINIMAL_AUTHOR_LAYOUT]: (
    <>
      <circle cx="45.73" cy="56.31" r="29.7" />
      <rect x="88.03" y="37.61" rx="0" ry="0" width="115" height="9" />
      <rect x="89.03" y="55.61" rx="0" ry="0" width="98" height="8" />
      <rect x="21.03" y="97.61" rx="0" ry="0" width="397.44" height="33" />
      <rect x="23.03" y="135.61" rx="0" ry="0" width="351.08" height="13.75" />
      <rect x="22.03" y="155.61" rx="0" ry="0" width="118" height="8" />
    </>
  ),
  [layouts.MINIMAL_LAYOUT]: (
    <>
      <rect x="14.03" y="3.61" rx="0" ry="0" width="472" height="35" />
      <rect x="14.03" y="50.61" rx="0" ry="0" width="324" height="11" />
      <rect x="16.03" y="72.61" rx="0" ry="0" width="290" height="10" />
    </>
  ),
};

export const sizes = {
  [layouts.MINIMAL_LAYOUT]: 100,
  [layouts.MINIMAL_AUTHOR_LAYOUT]: 200,
  [layouts.HORIZONTAL_REVERSE_LAYOUT]: 300,
  [layouts.HORIZONTAL_LAYOUT]: 300,
  [layouts.VERTICAL_LAYOUT]: 300,
};
