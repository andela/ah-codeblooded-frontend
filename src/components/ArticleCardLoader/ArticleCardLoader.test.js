import React from 'react';
import { shallow } from 'enzyme';
import ArticleCardLoader from './index';
import layouts from '../ArticleCard/layouts';

describe('<ArticleCardLoader/> component', () => {
  it('renders an loader with a particular layout', () => {
    const wrapper = shallow(<ArticleCardLoader layout={layouts.VERTICAL_LAYOUT} />);
    expect(wrapper).toMatchSnapshot();
  });
});
