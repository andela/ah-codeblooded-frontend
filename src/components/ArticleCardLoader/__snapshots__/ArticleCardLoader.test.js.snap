// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleCardLoader/> component renders an loader with a particular layout 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticleCardLoader
    layout="VERTICAL_LAYOUT"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <React.Fragment>
        <rect
          height="120"
          rx="0"
          ry="0"
          width="458"
          x="19.03"
          y="76.61"
        />
        <circle
          cx="48.44"
          cy="42.02"
          r="24.41"
        />
        <rect
          height="12"
          rx="0"
          ry="0"
          width="123"
          x="83.03"
          y="23.61"
        />
        <rect
          height="8"
          rx="0"
          ry="0"
          width="83"
          x="84.03"
          y="42.61"
        />
        <rect
          height="31.05"
          rx="0"
          ry="0"
          width="397"
          x="21.03"
          y="208.61"
        />
        <rect
          height="6"
          rx="0"
          ry="0"
          width="287"
          x="23.03"
          y="255.61"
        />
        <rect
          height="6"
          rx="0"
          ry="0"
          width="281"
          x="23.03"
          y="275.61"
        />
      </React.Fragment>,
      "height": 300,
      "primaryColor": "#cccccc",
      "rtl": true,
      "secondaryColor": "#ecebeb",
      "speed": 3,
      "width": 500,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <rect
            height="120"
            rx="0"
            ry="0"
            width="458"
            x="19.03"
            y="76.61"
          />,
          <circle
            cx="48.44"
            cy="42.02"
            r="24.41"
          />,
          <rect
            height="12"
            rx="0"
            ry="0"
            width="123"
            x="83.03"
            y="23.61"
          />,
          <rect
            height="8"
            rx="0"
            ry="0"
            width="83"
            x="84.03"
            y="42.61"
          />,
          <rect
            height="31.05"
            rx="0"
            ry="0"
            width="397"
            x="21.03"
            y="208.61"
          />,
          <rect
            height="6"
            rx="0"
            ry="0"
            width="287"
            x="23.03"
            y="255.61"
          />,
          <rect
            height="6"
            rx="0"
            ry="0"
            width="281"
            x="23.03"
            y="275.61"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "height": "120",
            "rx": "0",
            "ry": "0",
            "width": "458",
            "x": "19.03",
            "y": "76.61",
          },
          "ref": null,
          "rendered": null,
          "type": "rect",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "cx": "48.44",
            "cy": "42.02",
            "r": "24.41",
          },
          "ref": null,
          "rendered": null,
          "type": "circle",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "height": "12",
            "rx": "0",
            "ry": "0",
            "width": "123",
            "x": "83.03",
            "y": "23.61",
          },
          "ref": null,
          "rendered": null,
          "type": "rect",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "height": "8",
            "rx": "0",
            "ry": "0",
            "width": "83",
            "x": "84.03",
            "y": "42.61",
          },
          "ref": null,
          "rendered": null,
          "type": "rect",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "height": "31.05",
            "rx": "0",
            "ry": "0",
            "width": "397",
            "x": "21.03",
            "y": "208.61",
          },
          "ref": null,
          "rendered": null,
          "type": "rect",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "height": "6",
            "rx": "0",
            "ry": "0",
            "width": "287",
            "x": "23.03",
            "y": "255.61",
          },
          "ref": null,
          "rendered": null,
          "type": "rect",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "height": "6",
            "rx": "0",
            "ry": "0",
            "width": "281",
            "x": "23.03",
            "y": "275.61",
          },
          "ref": null,
          "rendered": null,
          "type": "rect",
        },
      ],
      "type": Symbol(react.fragment),
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <React.Fragment>
          <rect
            height="120"
            rx="0"
            ry="0"
            width="458"
            x="19.03"
            y="76.61"
          />
          <circle
            cx="48.44"
            cy="42.02"
            r="24.41"
          />
          <rect
            height="12"
            rx="0"
            ry="0"
            width="123"
            x="83.03"
            y="23.61"
          />
          <rect
            height="8"
            rx="0"
            ry="0"
            width="83"
            x="84.03"
            y="42.61"
          />
          <rect
            height="31.05"
            rx="0"
            ry="0"
            width="397"
            x="21.03"
            y="208.61"
          />
          <rect
            height="6"
            rx="0"
            ry="0"
            width="287"
            x="23.03"
            y="255.61"
          />
          <rect
            height="6"
            rx="0"
            ry="0"
            width="281"
            x="23.03"
            y="275.61"
          />
        </React.Fragment>,
        "height": 300,
        "primaryColor": "#cccccc",
        "rtl": true,
        "secondaryColor": "#ecebeb",
        "speed": 3,
        "width": 500,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <rect
              height="120"
              rx="0"
              ry="0"
              width="458"
              x="19.03"
              y="76.61"
            />,
            <circle
              cx="48.44"
              cy="42.02"
              r="24.41"
            />,
            <rect
              height="12"
              rx="0"
              ry="0"
              width="123"
              x="83.03"
              y="23.61"
            />,
            <rect
              height="8"
              rx="0"
              ry="0"
              width="83"
              x="84.03"
              y="42.61"
            />,
            <rect
              height="31.05"
              rx="0"
              ry="0"
              width="397"
              x="21.03"
              y="208.61"
            />,
            <rect
              height="6"
              rx="0"
              ry="0"
              width="287"
              x="23.03"
              y="255.61"
            />,
            <rect
              height="6"
              rx="0"
              ry="0"
              width="281"
              x="23.03"
              y="275.61"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "height": "120",
              "rx": "0",
              "ry": "0",
              "width": "458",
              "x": "19.03",
              "y": "76.61",
            },
            "ref": null,
            "rendered": null,
            "type": "rect",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "cx": "48.44",
              "cy": "42.02",
              "r": "24.41",
            },
            "ref": null,
            "rendered": null,
            "type": "circle",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "height": "12",
              "rx": "0",
              "ry": "0",
              "width": "123",
              "x": "83.03",
              "y": "23.61",
            },
            "ref": null,
            "rendered": null,
            "type": "rect",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "height": "8",
              "rx": "0",
              "ry": "0",
              "width": "83",
              "x": "84.03",
              "y": "42.61",
            },
            "ref": null,
            "rendered": null,
            "type": "rect",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "height": "31.05",
              "rx": "0",
              "ry": "0",
              "width": "397",
              "x": "21.03",
              "y": "208.61",
            },
            "ref": null,
            "rendered": null,
            "type": "rect",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "height": "6",
              "rx": "0",
              "ry": "0",
              "width": "287",
              "x": "23.03",
              "y": "255.61",
            },
            "ref": null,
            "rendered": null,
            "type": "rect",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "height": "6",
              "rx": "0",
              "ry": "0",
              "width": "281",
              "x": "23.03",
              "y": "275.61",
            },
            "ref": null,
            "rendered": null,
            "type": "rect",
          },
        ],
        "type": Symbol(react.fragment),
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
